### PR TITLE
Add `libhoney` package types

### PR DESCRIPTION
**Pull request by dcposch**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## libhoney ([<kbd>unpkg</kbd>](https://unpkg.com/browse/libhoney@latest/))
was missing the following properties:
1. once
1. on
1. listenerCount
1. errorMonitor
1. captureRejectionSymbol

<details>
<summary>as well as these 3 other properties...</summary>
<p>captureRejections, defaultMaxListeners, EventEmitter</p>
</details>


## Formatting

The following files are not formatted:
1. types/libhoney/tslint.json
1. types/libhoney/tsconfig.json
1. types/libhoney/libhoney-tests.ts
1. types/libhoney/index.d.ts

Consider running `npx dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 0c6afd7de6a0b3fb0346d32b2d37a26fa46968ea
</p>

File: types/libhoney/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,115 @@
+// Type definitions for libhoney 4.1
+// Project: https://github.com/honeycombio/libhoney-js#readme
+// Definitions by: dcposch <https://github.com/dcposch>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+export = libhoney;
+
+declare class libhoney {
```
Comment (peterblazejewicz) [inline types/libhoney/index.d.ts:8]: the symbol is named `Libhoney`:
https://github.com/honeycombio/libhoney-js/blob/main/src/libhoney.js#L74

Comment (dcposch) [reply types/libhoney/index.d.ts:8]: Good call, fixed.

**Comment by peterblazejewicz** [comment]
mind running `npx dprint fmt` as mentioned in bot comment?

**Comment by dcposch** [comment]
@peterblazejewicz done

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@dcposch thanks!

**Comment by dcposch** [comment]
Ready to merge

**Comment by jakebailey** [comment]
This PR mistakenly passed CI and did not have any CI checks run.

**Comment by jakebailey** [comment]
https://github.com/DefinitelyTyped/DefinitelyTyped/pull/67117

**Pull request merged**